dataikuapi/dss/projectdeployer.py

```diff
@@ -0,0 +1,451 @@
+        but not the actual health of the deployment
+
+        :rtype: dict
+        """
+        return self.light_status
+
+    def get_heavy(self):
+        """
+        Gets the 'heavy' (full) status. This returns a dictionary with various information about the deployment
+        :rtype: dict
+        """
+        return self.heavy_status
+
+    def get_health(self):
+        """Returns the health of this deployment as a string
+
+        :returns: HEALTHY if the deployment is working properly, various other status otherwise
+        :rtype: string
+        """
+        return self.heavy_status["health"]
+
+    def get_health_messages(self):
+        """Returns messages about the health of this deployment"""
+        return self.heavy_status["healthMessages"]
+
+###############################################
+# Published Project
+###############################################
+
+class DSSProjectDeployerProject(object):
+    """
+    A project on the Project Deployer
+
+    Do not create this directly, use :meth:`~dataikuapi.dss.projectdeployer.DSSProjectDeployer.get_project`
+    """
+    def __init__(self, client, project_key):
+        self.client = client
+        self.project_key = project_key
+
+    def id(self):
+        return self.project_key
+
+    def get_status(self):
+        """
+        Returns status information about this project. This is used mostly to get information about
+        which versions are available and which deployments are exposing this project
+
+        :rtype: dataikuapi.dss.projectdeployer.DSSProjectDeployerProjectStatus
+        """
+        light = self.client._perform_json("GET", "/project-deployer/projects/%s" % (self.project_key))
+        return DSSProjectDeployerProjectStatus(self.client, self.project_key, light)
+
+    def import_bundle(self, fp, origin_project_key=None, design_node_url=None, design_node_id=None):
+        """
+        Imports a new version for a project from a file-like object pointing
+        to a bundle Zip file.
+        :param string fp: A file-like object pointing to a bundle Zip file
+        :param string origin_project_key: The identifier of the Design node project where the bundle was created
+        :param string design_node_url: The URL of the Design node where the bundle was created
+        :param design_node_id: The identifier of the Design node where the bundle was created
+        """
+        if origin_project_key is None and design_node_url is None and design_node_id is None:
+            params = None
+        else:
+            params = {
+                "projectKey": origin_project_key,
+                "nodeId": design_node_id,
+                "url": design_node_url
+            }
+        return self.client._perform_empty("POST",
+                "/project-deployer/projects/%s/bundles" % (self.project_key), params=params, files={"file":fp})
```

Comment:
If you try to send any of the optional params, they won't be received on the DSS side: `designNodeInfo` will be `null` as the controller is looking for a request param with that name. One way to fix this is by doing `params={"designNodeInfo": json.dumps(params)}` (and `import json` at the top), which leads to a pretty ugly request. Maybe it's better to change the controller method to accept each of those three optional params individually? In that case, this code could remain unchanged, you would just need to build the `DesignNodeInfo` object manually in the controller method in the public API PR.

Reply:
Done in c1ed433, corresponding changes on the Public API endpoint in a86ba14